models/demos/falcon7b/tt/model_utils.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import tt_lib
+
+from models.utility_functions import torch2tt_tensor, pad_by_zero
+
+
+def get_weights_cached(
+    devices,
+    model_config,
+    tt_cache_path,
+    weight_cache_str,
+    weight_config_str,
+    weights_to_cache,
+    overwrite=False,
+    padzero=False,
+):
+    """Load cached weights and duplicate per device. Store if not cached."""
+    if (
+        not overwrite
+        and (tt_cache_path / f"{weight_cache_str}_{model_config[f'{weight_config_str}_DTYPE'].name}.bin").exists()
+    ):
+        # Load cached weights
+        weights_host = tt_lib.tensor.load_tensor(
```

Comment:
More of an FYI since you're free to do what you want in your test/model-side code but 
 @arakhmati recently made `ttnn.Tensor` and `tt_lib.Tensor` to be one and the same. So you should be able to replace the load tensor from cache logic with `ttnn.as_tensor` and get some of this logic for free.

Reply:
Thanks Joseph, good point. I'll update that in an upcoming PR since this one is already quite large.